<launch>
	<arg name="veh"/>
	<arg name="exercise_name" default="1-1"/>
	<arg name="config" default="baseline"/>
	<arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman"/>
	<arg name="param_file_name_wheels_driver_node" default="default" doc="Specify a param file. ex:megaman"/>
	<!--arg name="local" default="false" doc="true to launch locally on laptop. false to launch of vehicle"/-->
	<arg name="pkg_name" default="lane_control" doc="name of the package"/>
	<!--arg name="pkg_name_wheels_driver_node" default="dagu_car" doc="name of the package"/-->
	<arg name="node_name" default="lane_exercise_controller_node" doc="name of the node"/>
	<!--arg name="node_name_wheels_driver_node" default="wheels_driver_node" doc="name of the node"/-->

	<group ns="$(arg veh)">
	    <!-- Local -->
	    <node pkg="$(arg pkg_name)" type="$(arg node_name).py" name="$(arg node_name)" output="screen" required="true">
			<remap from="lane_exercise_controller_node/lane_pose" to="lane_controller_node/lane_pose"/>
			<remap from="lane_exercise_controller_node/switch" to="lane_controller_node/switch"/>
			<remap from="lane_exercise_controller_node/at_stop_line" to="lane_controller_node/at_stop_line"/>
			<remap from="lane_exercise_controller_node/fsm_mode" to="lane_controller_node/fsm_mode"/>
			<remap from="lane_exercise_controller_node/car_cmd" to="lane_controller_node/car_cmd"/>
			<remap from="lane_exercise_controller_node/veh_car_cmd" to="vehicle_avoidance_control_node/car_cmd"/>

			<param name="exercise_name" value="$(arg exercise_name)"/>
        <rosparam command="load" file="$(find lane_control)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
	    </node>
      <!-- Remote >
		    <include unless="$(arg local)" file="$(find duckietown)/machines"/>
	    <node unless="$(arg local)" machine="$(arg veh)" pkg="$(arg pkg_name)" type="$(arg node_name).py" name="$(arg node_name)" output="screen" clear_params="true" required="true">
			<remap from="lane_exercise_controller_node/lane_pose" to="lane_controller_node/lane_pose"/>
			<remap from="lane_exercise_controller_node/switch" to="lane_controller_node/switch"/>
			<remap from="lane_exercise_controller_node/at_stop_line" to="lane_controller_node/at_stop_line"/>
			<remap from="lane_exercise_controller_node/fsm_mode" to="lane_controller_node/fsm_mode"/>
			<remap from="lane_exercise_controller_node/car_cmd" to="lane_controller_node/car_cmd"/>
			<remap from="lane_exercise_controller_node/veh_car_cmd" to="vehicle_avoidance_control_node/car_cmd"/>

			<param name="exercise_name" value="$(arg exercise_name)"/>
	        <rosparam command="load" file="$(find duckietown)/config/$(arg config)/$(arg pkg_name_wheels_driver_node)/$(arg node_name_wheels_driver_node)/$(arg param_file_name_wheels_driver_node).yaml"/>
	    </node-->
	</group>

	<!-- Publication -->
	<!-- ~wheels_control: duckietown_msgs/WheelsCmd. Wheels command for duckiebot. -->

	<!-- Subscription -->
	<!-- ~lane_pose: duckietown_msgs/LanePose. LanePose from the lane filter -->


</launch>
